pybop/models/empirical/base_ecm.py

```diff
@@ -4,10 +4,77 @@
 class ECircuitModel(BaseModel):
     """
     Overwrites and extends `BaseModel` class for circuit-based PyBaMM models.
+
+    Parameters
+    ----------
+    pybamm_model : pybamm.BaseModel
+        A subclass of the pybamm Base Model.
+    name : str, optional
+        The name for the model instance, defaulting to "Electrochemical Base Model".
+    parameter_set : pybamm.ParameterValues or dict, optional
+        The parameters for the model. If None, default parameters provided by PyBaMM are used.
+    geometry : dict, optional
+        The geometry definitions for the model. If None, default geometry from PyBaMM is used.
+    submesh_types : dict, optional
+        The types of submeshes to use. If None, default submesh types from PyBaMM are used.
+    var_pts : dict, optional
+        The discretization points for each variable in the model. If None, default points from PyBaMM are used.
+    spatial_methods : dict, optional
+        The spatial methods used for discretization. If None, default spatial methods from PyBaMM are used.
+    solver : pybamm.Solver, optional
+        The solver to use for simulating the model. If None, the default solver from PyBaMM is used.
+    **model_kwargs : optional
+        Valid PyBaMM model option keys and their values. For example,
+        build : bool, optional
+            If True, the model is built upon creation (default: False).
+        options : dict, optional
+            A dictionary of options to customise the behaviour of the PyBaMM model.
     """
 
-    def __init__(self, name, parameter_set):
-        super().__init__(name, parameter_set)
+    def __init__(
+        self,
+        pybamm_model,
+        name,
+        parameter_set=None,
+        geometry=None,
+        submesh_types=None,
+        var_pts=None,
+        spatial_methods=None,
+        solver=None,
+        **model_kwargs,
+    ):
+        super().__init__(name=name, parameter_set=parameter_set)
+
+        model_options = dict(
+            build=False,
+            options=None,
```

Comment:
I don't believe we need to construct `options` here. Is it done to signpost to developers that it is an available key?